src/main/modules/oidc/index.ts

```diff
@@ -64,11 +64,14 @@ export class OidcMiddleware {
 
         return next();
       }
-      res.redirect('/login');
-    });
 
+      if (req.xhr) {
+        res.status(302).send({ url: '/login' });
+      } else {
+        res.redirect('/login');
+      }
+    });
   }
```

Comment:
This seems to be working as expected @agraham01 - I'm testing by invalidating the session through removing the session cookie, and it does lead me to being sent to the login page if I try commit a change without a valid session.